src/personal_api_keys/PersonalApiKeyForm.tsx

```diff
@@ -0,0 +1,281 @@
+import { ModalDeleteContent } from '../components/ModalDeleteContent'
+import FormActionButtons from '../components/FormActionButtons';
+
+
+interface Props {
+  currentRecord?: any;
+  allPersonalApiKeys?: [];
+};
+interface PropsFromDispatch {
+  addNotification: (message: string | number, timeout: number) => void
+};
+interface RouteParams {
+  uuid: string;
+};
+
+const PersonalApiKeyFormModel: React.FC<Props & PropsFromDispatch & RouteComponentProps<RouteParams>> = (props) => {
+  const { currentRecord, allPersonalApiKeys } = props;
+  // @ts-ignore
+  const recordWithFtpExists = ((allPersonalApiKeys || []).filter(record=>(record.scope+'').includes("ftp:readwrite"))).length > 0;
+  const [apiKeyString, setApiKeyString ] = useState("");
+  const [showDeleteModal, setShowDeleteModal ] = useState(false);
+
+  const initialValues = {
+    name: (currentRecord && currentRecord.name) || '',
+    scopeWildcardReadWrite: (currentRecord && currentRecord.scope && (currentRecord.scope+'').includes('*:readwrite')) || true,
+    scopeFtpReadWrite: (currentRecord && currentRecord.scope && (currentRecord.scope+'').includes('ftp:readwrite')) || false,  
+  };
+
+  const getScopeStringFromValues = (values:any) => {
+    let scopeArr = [];
+    if (values.scopeWildcardReadWrite) {
+      scopeArr.push('*:readwrite');
+    }
+    if (values.scopeFtpReadWrite) {
+      scopeArr.push('ftp:readwrite');
+    }
+
+    return scopeArr.join(" ");
+  }
+
+  const onDelete = () => {
+    const body = {};
+
+    currentRecord && fetch(`/api/v4/personalapikeys/${currentRecord.prefix}/`, {
+      credentials: 'same-origin',
+      method: 'DELETE',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify(body)
+    })
+      .then(data => {
+        const status = data.status;
+        if (status === 204) {
+          props.addNotification('Success! API key deleted', 2000);
+          props.history.push('/personal_api_keys/');
+        } else {
+          props.addNotification(status, 2000);
+          console.error(data);
+        };
+      })
+      .catch(console.error);
+  }
+
+  const onSubmit = (values: Values) => {
+    const body = {
+      name: values.name,
+      scope: getScopeStringFromValues(values),
+    };
+
+    fetch(`/api/v4/personalapikeys/`, {
+      credentials: 'same-origin',
+      method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify(body)
+    })
+      .then(data => {
+        const status = data.status;
+        if (status === 201) {
+          // setting both modal and snackbar message is a bit too much so I comment snackbar message out
+          // props.addNotification('Success! Personal API key generated', 2000);
+          data.json().then((record: any)=>{
+            setApiKeyString(record.key+'');
+          })
+          // do not redirect because we need to show the api key modal
+          // props.history.push('/personal_api_keys/');
```

Comment:
I think commented out code like these can be safely deleted from the code base.

Reply:
done